src/prefect/client/client.py

```diff
@@ -241,6 +219,36 @@ def post(
         else:
             return {}
 
+    def _init_tenant(self) -> None:
+        """
+        Init the tenant to contact the server
+
+        If your backend is set to cloud the tenant will be read from: $HOME/.prefect/settings.toml
+
+        For the server backend it will try to retrieve the default tenant. If the server is
+        protected with auth like BasicAuth do not forget to `attach_headers` before any call.
+        """
+        if prefect.config.backend == "cloud":
+            if not self._api_token:
+                # if no api token was passed, attempt to load state from local storage
+                settings = self._load_local_settings()
+                self._api_token = settings.get("api_token")
+
+                if self._api_token:
+                    self._active_tenant_id = settings.get("active_tenant_id")
+                if self._active_tenant_id:
+                    try:
+                        self.login_to_tenant(tenant_id=self._active_tenant_id)
+                    except AuthorizationError:
+                        # if an authorization error is raised, then the token is invalid and should
+                        # be cleared
+                        self.logout_from_tenant()
+        else:
+            # TODO: Separate put this functionality and clean up initial tenant access handling
```

Comment:
Is this todo relevant now? Perhaps during this change `_init_tentant` should be split into its two separate modes

Reply:
I don't know the scope of it. So I'll let the decision to the maintainer  it :+1:

Reply:
This PR actually addresses this TODO so it can be removed 🙂